app/controllers/hotels_controller.rb

```diff
@@ -35,16 +37,26 @@ def destroy
 
   private
 
-  def authorize_resource
-    authorize hotel
+  def decorated_hotels
+    HotelDecorator.decorate_collection(filtered_hotels, context: { coordinates: current_coordinates })
   end
 
   def filtered_hotels
-    ::FilteredHotels.new(Hotel.includes(:city).all, filter_params).all.page(params[:page])
+    ::FilteredHotels.new(fetch_hotels, filter_params).all
+  end
+
+  def fetch_hotels
+    Hotel.near(current_coordinates, DISTANCE).includes(:city).page(params[:page])
+  end
+
+  def authorize_resource
+    authorize hotel
   end
 
   def filter_params
-    params.fetch(:hotel, {}).permit(:search, :stars, :min_rating, :max_rating).to_h
+    params.fetch(:hotel, {})
```

Comment:
I think that you can add `near` method into query object, instead of `radius` and `coordinates`.
```
def fetch_hotels
   Hotel.includes(:city).page(params[:page])
end

def filter_params
  hotel_params.permit(:search, :stars, :min_rating, :max_rating).merge(near: near_params)
end

def near_params
  {
    radius: hotel_params.permit(:radius),
    coordinates: current_coordinates
  }
end

def hotel_params
   params.fetch(:hotel, {})
end
```